packages/jest-runtime/src/index.ts

```diff
@@ -81,7 +81,11 @@ const defaultTransformOptions: InternalModuleOptions = {
 type InitialModule = Partial<Module> &
   Pick<Module, 'children' | 'exports' | 'filename' | 'id' | 'loaded'>;
 type ModuleRegistry = Map<string, InitialModule | Module>;
-type ResolveOptions = Parameters<typeof require.resolve>[1];
+
+type ResolveOptions = Parameters<typeof require.resolve>[1] & {
+  outsideJestVm?: true;
+};
+const OUTSIDE_JEST_VM_PROTOCOL = 'outside-jest-vm:';
```

Comment:
nit: "jest-parent:" or "jest-main:" or "jest:".

Reply:
Hmm, parent/main sound a lot like it's about main process vs workers, when it's really about Node require vs jest-runtime require (a distinction that also applies in runInBand) so I'm not sure I find any of these clearer :/ Happy to change to something that is prettier but still unambigious

Reply:
I think just `jest-internal:` as a protocol would work well.

Reply:
Settled for `jest-main` after all, don't think it's that ambigious actually